plrustc/plrustc/src/lints.rs

```diff
@@ -149,12 +150,66 @@ impl EarlyLintPass for PlrustAsync {
     }
 }
 
+declare_lint!(
+    pub(crate) PLRUST_LEAKY,
+    Allow,
+    "Disallow use of `{Box,Vec,String}::leak`, `mem::forget`, and similar functions",
+);
+
+declare_lint_pass!(PlrustLeaky => [PLRUST_LEAKY]);
+
+impl<'tcx> LateLintPass<'tcx> for PlrustLeaky {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &hir::Expr) {
+        let paths: &[&[&str]] = &[
+            &["alloc", "boxed", "Box", "leak"],
+            &["alloc", "vec", "Vec", "leak"],
+            &["alloc", "string", "String", "leak"],
+            &["core", "mem", "forget"],
+        ];
+        for &path in paths {
+            if does_expr_call_path(cx, expr, path) {
+                cx.lint(
+                    PLRUST_LEAKY,
+                    "Leaky functions are forbidden in PL/Rust",
+                    |b| b.set_span(expr.span),
+                );
+            }
+        }
+    }
+}
+
+fn does_expr_call_path(cx: &LateContext<'_>, expr: &Expr<'_>, segments: &[&str]) -> bool {
```

Comment:
This name is still sort of inaccurate. It's more "is it a path expr or method call for the path", but that's what you want most of the time (because we should catch `let x = Vec::leak; x(vec![a, b, c])`)

Reply:
To be clear, that case is caught. I just added a test for it.

Reply:
Oh okay!